### PR TITLE
LBGG[Nuxt3]: SiteNavbar

### PR DESCRIPTION
**Pull request by erunks**
## What

Ported all the components that make up the `<SiteNavbar />` from `Vue2` to `Vue3`

### Steps for testing
- Pull the code
- `pnpm install && pnpm dev` to get things running
- Check the modals respond and the inputs work

#### SiteNavbar
| • | Mobile | Tablet | Desktop |
| --- | --- | --- | --- |
| Before | image | image | image |
| After | image | image | image |

#### Modals
| • | Mobile | Tablet | Desktop |
| --- | --- | --- | --- |
| Log In Modal | image | image | image |
| Sign Up Modal | image | image | image |

### CODE REVIEW

File: components/blocks/SiteFooter/SiteFooter.test.ts [hunk 1]
```diff
@@ -11,6 +11,6 @@ describe('<SiteFooter />', () => {
   it('renders correctly with the copyright info', () => {
     const { getByText } = stubbedRender(SiteFooter)
 
-    expect(getByText('Copyright 2021')).toBeInTheDocument()
+    expect(getByText('Copyright 2022')).toBeInTheDocument()
```
Comment (zysim) [inline components/blocks/SiteFooter/SiteFooter.test.ts:14]: We could add a build step that automatically updates this. Like a `sed` or `rg` call or something

File: components/blocks/cards/LogInCard/LogInCard.vue [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  modal?: boolean
+}
+
+interface LoginCardState {
+  email: string
+  password: string
+  showPassword: boolean
+}
+
+const emit = defineEmits<LoginCardEmits>()
```
Comment (buffet-time) [inline components/blocks/cards/LogInCard/LogInCard.vue:25]: It's better if we move the emits definitions into the defineEmits itself imho.

File: components/elements/buttons/BaseButton/BaseButton.vue [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<script setup lang="ts" />
```
Comment (buffet-time) [inline components/elements/buttons/BaseButton/BaseButton.vue:1]: I think currently this file doesn't need even an empy `script setup` block

File: components/elements/buttons/HideShowPassword/HideShowPassword.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+function toggleHidden() {
+  state.hidden = !state.hidden
+}
+</script>
+<script lang="ts">
+export default {
+  name: 'HideShowPassword',
+  components: {
+    BaseButton,
```
Comment (buffet-time) [inline components/elements/buttons/HideShowPassword/HideShowPassword.vue:23]: Why is this here? We're importing it in the setup block so this is not needed.

File: components/elements/modals/BaseModal/BaseModal.vue [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+</script>
+
+<template>
+  <div class="modal">
+    <div class="modal__wrapper">
+      <div class="modal__container">
+        <button
+          class="modal-fade"
+          type="button"
+          @click.self="$emit('close')"
```
Comment (buffet-time) [inline components/elements/modals/BaseModal/BaseModal.vue:13]: There are no emits defined in the script here

File: components/blocks/SiteFooter/SiteFooter.test.ts (hunk 1 above)
Comment (erunks) [reply components/blocks/SiteFooter/SiteFooter.test.ts]: Went ahead and just had it pull the current year from a new date object instead

Comment (zysim) [reply components/blocks/SiteFooter/SiteFooter.test.ts]: That definitely works too ha
